src/Utils/Hooks/useMap.ts

```diff
@@ -0,0 +1,89 @@
+import React, {useEffect,useRef,MutableRefObject} from 'react'
+import {renderToString} from 'react-dom/server';
+import H from "@here/maps-api-for-javascript"
+
+interface ICenterMap{
+    lat:number;
+    lng:number;
+}
+
+/**
+ * renders a map in a container div, based on a coordinates and a level of zoom. apikey for the sdk is needed.
+ * @param mapContainer {React.RefObject<HTMLDivElement>} - the reference to the container div for the map 
+ * @param apikey {string} - the api key for the sdk
+ * @param center {ICenterMap} - the map coordinates (its center)
+ * @param zoom {number} - the level of zoom applied
+ * @returns {MutableRefObject<H.Map | undefined>} a reference to the map instance created
+ */
+export const useMap=(mapContainer:React.RefObject<HTMLDivElement>,apikey:string,center:ICenterMap,zoom:number):MutableRefObject<H.Map | undefined>=> {
+    const mapInstance=useRef<H.Map>()
+    const platform=useRef<H.service.Platform>()
+    
+    useEffect(()=>{
+        if(!platform.current)
+            platform.current = new H.service.Platform({
+                apikey
+            })
+    
+        const defaultLayers = platform.current.createDefaultLayers();
+    
+        if(!mapInstance.current&&mapContainer.current?.clientWidth)
+            mapInstance.current=new H.Map(
+                mapContainer.current,
+                defaultLayers.vector.normal.map,
+                {
+                    center,
+                    zoom,
+                    pixelRatio: window.devicePixelRatio || 1
+                }
+            )
+
+        return ()=>{
+            if(mapInstance.current) {
+                mapInstance.current.dispose()
+                mapInstance.current=undefined
+            }
+        }
+    },[mapContainer.current,mapContainer.current?.clientWidth]) 
+
+    return mapInstance
+}
+
+/**
+ * puts a marker in a map in certain coordinates
+ * @param map {MutableRefObject<H.Map | undefined>} - the map instance reference
+ * @param mapCoordinates {ICenterMap} - the coordinates on the map where to place the mark
+ * @param element {React.ReactElement} - the rendered react element which is the mark
+ */
+export const useMapMarker=(map:MutableRefObject<H.Map | undefined>,mapCoordinates:ICenterMap,element:React.ReactElement):void=>{
+    useEffect(()=>{
```

Comment:
Why not use layouts effect?

Reply:
I tryied and there is no much difference in the end result so I suppose I will stay with useEffect